src/richie/apps/courses/templates/courses/cms/fragment_course_glimpse.html

```diff
@@ -45,7 +45,7 @@
           </p>
           {% if course_state.call_to_action %}
           <div class="course-glimpse__footer__cta">
-              <button class="button">{{ course_state.call_to_action|capfirst }}</button>
```

Comment:
I disagree with this change.

Generally, as a rule, if it looks like a button it should be a `<button>` not a `<span class="button">`.

I understand how this is is specific situation where it is inside an `<a>` but I'd need something more convincing than an HTML validator warning to override the rule I state above.

Reply:
This is not a warning, it is an error. And this is not "an HTML validator", this is THE validator, we are talking about Nu Html Checker which is the official W3C validator which strictly follow the HTML reference.
 
The button element in this context has only a layout meaning. Semantically and accessibility talking it was wrong. Alike you don't put a `<a>` in another  `<a>`, you must not put  `<button>` in  `<a>`.

The choice of `<span>` is because it's the less annoying element since `.button` class does everything itself, but we may use a simple `<div>` if it troubles you (but not `<p>` since it possibly involves some default margin/padding/font settings from bootstrap typography.

Reply:
To be completely frank, I think HTML validation only has value when it allows us to detect issues that actual users (incl. assistive technology users) face. Beyond that, in my opinion HTML being valid or not is not something to be concerned about.

I searched for data about whether it's better for accessibility tools to keep a real `<button>` here or resort to the `<span>` workaround and could not find anything. Do you have anything on this?

Reply:
HTML validation is relevant since it valid browser doest not switch in quirk mode to automatically resolve wrong html.

I don't have any links about `<button>` is better than a `<span>` in our context, since the button was here only for layout interest since it lives inside a link and so the whole glimpse is clickable, it does not change anything to a span or a div or something else, it even not have any accessibility issue, it's just cosmetic.

`button` element has been introduced to use an element which have default browser style and implicit aria role just like a input button that must be inside a form, so the `button` element can be used outside a form.

Reply:
I'm still not convinced about the relevance of HTML validation, and we don't have any information on what's the better choice here for accessibility. It's not very important since everything is wrapped in an `<a>` anyway.

Let's merge this 🤝